src/main/haskell/kore/src/Kore/Step/OrOfExpandedPattern.hs

```diff
@@ -483,3 +490,16 @@ toExpandedPattern (MultiOr patts) =
                 , predicate = makeTruePredicate
                 , substitution = mempty
                 }
+
+{-| Transforms an 'OrOfPredicate' into a 'Predicate'. -}
+toPredicate
```

Comment:
It was caused by any simplification that would return an `\or` inside a `Predicate`, because this gets unpacked into a `MultiOr`, then re-packed, then unpacked... by the simplifier. Once I finish my current task of implementing branching in the predicate simplifier it won't matter because we will never re-pack such things.